### PR TITLE
https://github.com/mikaelbr/node-notifier/issues/99

### LINKED ISSUE #99: Windows 8.1 does not support `file:///xxx` protocol icon paths
When using a `file:///xxx` protocol path for the icon, it will just fail on Windows 8.1. I do see that documentation always says to use absolute paths and doesn't mention file protocol paths; It was an oversight on my part because I was refactoring an [nw.js app using `window.Notifcation` where `file:///` is needed](https://github.com/nwjs/nw.js/wiki/Notification#toast-notification-requires-node-webkit--v011).

Platform test results when using `file:///xxx` paths (`node-notifier@4.5.0`):

 - OSX: Works
 - Windows 10: Works
 - Windows 8.1: Does not work

Example:

```js
var notifier = require('node-notifier');

notifier.notify({
  'title': 'Heya',
  'message': 'foo bar',
  'icon': 'file:///C:/some-project/mlm.png'
  //'icon': 'file:///C:\\some-project\\mlm.png'
});
```

### PR DESCRIPTION
**Pull request by gucong3000**
fix #99 & add test for this

### CODE REVIEW

File: test/toaster.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  after(function () {
+    utils.fileCommand = this.original;
+    os.type = this.originalType;
+    os.arch = this.originalArch;
+    os.release = this.originalRelease;
+  });
+
+  it('should parse file protocol URL', function (done) {
+    utils.fileCommand = function (notifier, argsList, callback) {
+      argsList[1].should.eql(path.resolve("C:\\node-notifier\\test\\fixture\\coulson.jpg"));
```
Comment (mikaelbr) [inline test/toaster.js]: This should probably use `path.sep` or even better `path.join`?

**Comment by mikaelbr** [comment]
Brilliant! Thanks again. I haven't gotten around to prioritize this yet. 

Could you also add a test without `file://` schema?

**Comment by gucong3000** [comment]
@mikaelbr ok

**Comment by gucong3000** [comment]
Done.
Test for:
-   'http://csscomb.com/img/csscomb.jpg'
-   `path.join(__dirname, 'fixture', 'coulson.jpg')`

**Comment by mikaelbr** [comment]
Looks good! Thanks 🎉

**Pull request merged**